jdaviz/app.py

```diff
@@ -1304,14 +1309,33 @@ def is_there_overlap_spectral_subset(self, subset_name):
         Returns True if the spectral subset with subset_name has overlapping
         subregions.
         """
-        spectral_region = self.get_subsets(subset_name, spectral_only=True)
-        if not spectral_region or len(spectral_region) < 2:
+        spectral_region = self.get_subsets(subset_name, simplify_spectral=False, spectral_only=True)
```

Comment:
Because I have to move the "merge overlapping regions" calls into `get_subsets` when `simplify_spectral=True`, I have to force `simplify_spectral=False` here so this check is still correct.